### PR TITLE
Navigation: Fix overflowing menu name in the navigation selector dropdown

### PR DESCRIPTION
**Pull request by yuliyan**
## What?
This PR fixes the overflowing text in the navigation selector dropdown for menus with longer names.

## Why?

It resolves the UI bug with the navigation selector dropdown reported in https://github.com/WordPress/gutenberg/issues/45294.

## How?

This PR wraps the text for the selected menu within the navigation selector dropdown toggle, so that flex and text CSS properties can be applied to avoid the overflowing.

## Testing Instructions

1. Add a new Navigation block.
2. Create a menu with a really long name.
3. Select the menu from the dropdown.
4. Check if the menu name within the dropdown toggle button gets truncated instead of overflowing outside of the button.

## Screenshots or screencast 
**Before:**
CleanShot 2022-11-09 at 11 24 02

**After:**
CleanShot 2022-11-09 at 11 23 10

### CODE REVIEW

**Review by getdave — COMMENTED** [review]
Thanks for the PR. I tested it to the extreme and this is what I saw

Screen Shot 2022-11-09 at 11 43 16

The `<LinkControl>` component does a lot with overflow and ellipsis so perhaps there are some learnings we could take from that?

**Comment by yuliyan** [comment]
@getdave, thanks for the review. It looks like I had accidentally deleted one of the properties before committing the changes.

I've added it back so it should now be working as expected. Could you pull the changes and take another look?

File: packages/block-library/src/navigation/editor.scss [hunk 1]
```diff
@@ -158,7 +157,7 @@ $colors-selector-size: 22px;
 		min-width: $colors-selector-size;
 		height: $colors-selector-size;
 		min-height: $colors-selector-size;
-		line-height: ( $colors-selector-size - 2 );
+		line-height: ($colors-selector-size - 2);
```
Comment (getdave) [inline packages/block-library/src/navigation/editor.scss]:
```suggestion
		line-height: ( $colors-selector-size - 2 );
```

This change seems unrelated.

File: packages/block-library/src/navigation/editor.scss [hunk 2]
```diff
@@ -202,7 +201,9 @@ $color-control-label-height: 20px;
 // This should be a temporary fix, to be replaced by improvements to
 // the sibling inserter. Or by the dropdown appender that is used in Group, instead of the "Button block appender".
 // See https://github.com/WordPress/gutenberg/issues/37572.
-.wp-block-navigation .wp-block + .block-list-appender .block-editor-button-block-appender {
+.wp-block-navigation
+.wp-block + .block-list-appender
+.block-editor-button-block-appender {
```
Comment (getdave) [inline packages/block-library/src/navigation/editor.scss]: Seems like another unrelated formatting change we should revert

File: packages/block-library/src/navigation/editor.scss [hunk 3]
```diff
@@ -389,7 +389,9 @@ $color-control-label-height: 20px;
 }
 
 // Keep as row for medium.
-.wp-block-navigation .components-placeholder.is-medium .components-placeholder__fieldset {
+.wp-block-navigation
+.components-placeholder.is-medium
+.components-placeholder__fieldset {
```
Comment (getdave) [inline packages/block-library/src/navigation/editor.scss]: And one more unrelated formatting change...

File: packages/block-library/src/navigation/editor.scss [hunk 4]
```diff
@@ -570,7 +571,8 @@ body.editor-styles-wrapper
 	padding: $grid-unit-10 $grid-unit-15;
 }
 
-.wp-block-navigation .wp-block-navigation__uncontrolled-inner-blocks-loading-indicator {
+.wp-block-navigation
+.wp-block-navigation__uncontrolled-inner-blocks-loading-indicator {
```
Comment (getdave) [inline packages/block-library/src/navigation/editor.scss]: And one more.

**Review by getdave — APPROVED** [review]
Tested and works as described. Thank you for the fix 👏 

Left some small changes that appeared unrelated which we should revert.

**Review by getdave — APPROVED** [review]
